crates/fuel-core/src/service/query.rs

```diff
@@ -99,11 +100,28 @@ struct StatusChangeState<'a> {
 }
 
 impl<'a> TxnStatusChangeState for StatusChangeState<'a> {
-    async fn get_tx_status(&self, id: Bytes32) -> StorageResult<Option<TxPoolTxStatus>> {
+    async fn get_tx_status(
+        &self,
+        id: Bytes32,
+        allow_preconfirmation: bool,
+    ) -> StorageResult<Option<TxPoolTxStatus>> {
         match self.db.get_tx_status(&id)? {
             Some(status) => Ok(Some(status.into())),
             None => {
                 let status = self.tx_status_manager.status(id).await?;
+                let status = status.map(|status| {
+                    // Filter out preconfirmation statuses if not allowed. Converting to submitted status
+                    // because it's the closest to the preconfirmation status.
+                    // Having `now()` as timestamp isn't ideal but shouldn't cause much inconsistency.
```

Comment:
Should we keep submitted timestamp when a tx is preconfirmed?

Reply:
Maybe we can open a followup discussion because it would require the emitter of preconfirmation (executor) to get something else than just a transaction but a transaction with some metadata